src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -4931,17 +4917,24 @@ void handleMemberInitializer(IMemberInitializerOperation memberInitializer)
                         _evalStack.Push(Visit(arrayReference.ArrayReference));
                         return (success: true, indicies);
 
-                    case OperationKind.DynamicIndexerAccess: // PROTOTYPE(dataflow): For now handle as invalid case to enable other test scenarios
-                    case OperationKind.None: // PROTOTYPE(dataflow): For now handle as invalid case to enable other test scenarios.
-                                             //                      see Microsoft.CodeAnalysis.CSharp.UnitTests.IOperationTests.ObjectCreationWithDynamicMemberInitializer_01
-                    case OperationKind.Invalid:
-                        return (success: false, arguments: ImmutableArray<IOperation>.Empty);
+                    case OperationKind.DynamicIndexerAccess:
+                        var dynamicIndexer = (IDynamicIndexerAccessOperation)instance;
+                        ImmutableArray<IOperation> arguments = dynamicIndexer.Arguments.SelectAsArray(argument =>
+                        {
+                            int captureId = VisitAndCapture(argument);
+                            return (IOperation)GetCaptureReference(captureId, argument);
+                        });
+                        _evalStack.Push(Visit(dynamicIndexer.Operation));
+                        return (success: true, arguments);
 
                     default:
-                        // PROTOTYPE(dataflow): Probably it will be more robust to handle all remaining cases as an invalid case.
-                        //                      Likely not worth it throwing for some edge error case. Asserting would be good though
-                        //                      so that tests could catch the things we though are impossible.
-                        throw ExceptionUtilities.UnexpectedValue(instance.Kind);
+                        // As in the assert in handleInitializer, this assert documents the operation kinds that we know go through this path,
+                        // and it is possible others go through here as well. If they are encountered, we simply need to ensure
+                        // that they don't have any interesting semantics in object or collection initialization contexts and add them to the
+                        // assert.
+                        Debug.Assert(instance.Kind == OperationKind.Invalid || instance.Kind == OperationKind.None ||
+                                     instance.Kind == OperationKind.DynamicMemberReference);
```

Comment:
>instance.Kind == OperationKind.DynamicMemberReference [](start = 37, length = 53)

Why are we considering  DynamicMemberReference a failure case, it looks like it should be treated as a success case. Otherwise, if there is a control flow in the value, we are likely to spill the DynamicMemberReference, which would have negative impact on the shape of the tree. Could you please add a test for a scenario like that. Something simple, just one assignment in the initializer with a coalesce on the right.